cmd/registry/cmd/compute/conformance.go

```diff
@@ -0,0 +1,208 @@
+// Copyright 2020 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compute
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/apigee/registry/cmd/registry/cmd/compute/conformance"
+	"github.com/apigee/registry/cmd/registry/core"
+	"github.com/apigee/registry/connection"
+	"github.com/apigee/registry/rpc"
+	"github.com/apigee/registry/server/names"
+	"github.com/spf13/cobra"
+	"google.golang.org/protobuf/proto"
+)
+
+func conformantsCommand(ctx context.Context) *cobra.Command {
+	var linter string
+	cmd := &cobra.Command{
+		Use:   "conformance",
+		Short: "Compute lint results for API specs",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			filter, err := cmd.Flags().GetString("filter")
+			if err != nil {
+				log.Fatalf("Failed to get filter from flags: %s", err)
+			}
+
+			client, err := connection.NewClient(ctx)
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+
+			name := args[0]
+
+			// Ensure that the provided argument is a spec
+			var specSegments []string
+			if specSegments = names.SpecRegexp().FindStringSubmatch(name); specSegments == nil {
+				log.Fatalf(
+					fmt.Sprintf(
+						"The provided argument %s does not match the regex of a spec",
+						name,
+					),
+				)
+			}
+			spec, err := names.ParseSpec(name)
+			if err != nil {
+				log.Fatalf("%s", err.Error())
+			}
+
+			projectSegments := []string{"projects", spec.ProjectID}
+
+			linterNameToLinter := make(map[string]conformance.Linter)
+			err = core.ListArtifacts(ctx, client, projectSegments, filter, true, func(artifact *rpc.Artifact) {
```

Comment:
Remind me the behavior of this method. Does this block on the network request and the callback?

Reply:
Yes it does. It makes a single call to get the artifacts via the client, and then loops over them an applies the provided callback function to each.

Reply:
Yes it does. It makes a single call to get the artifacts via the client, and then loops over them an applies the provided callback function to each.